data_model/src/evaluate.rs

```diff
@@ -1,105 +1,107 @@
 //! Implementations for Expression evaluation for different expressions.
-#![allow(
-    clippy::module_name_repetitions,
-    clippy::std_instead_of_core,
-    clippy::std_instead_of_alloc,
-    clippy::arithmetic_side_effects
-)]
-use eyre::Result;
-use iroha_data_model::{
+
+#[cfg(not(feature = "std"))]
+use alloc::{collections::BTreeMap, string::ToString};
+#[cfg(feature = "std")]
+use std::collections::BTreeMap;
+
+use crate::{
     expression::{prelude::*, Expression},
     isi::error::{InstructionExecutionFailure as Error, MathError},
     prelude::*,
-    query::error::FindError,
+    query::error::{FindError, QueryExecutionFailure},
 };
 
-use super::Evaluate;
-use crate::{prelude::ValidQuery, wsv::WorldStateView};
+/// Context of expression evaluation, holding (name, value) pairs for resolving identifiers
+pub trait Context: Clone {
+    /// Execute query against the current state of `Iroha`
+    ///
+    /// # Errors
+    ///
+    /// If query evaluation/execution fails
+    fn query(&self, query: &QueryBox) -> Result<Value, QueryExecutionFailure>;
```

Comment:
Okay, I see, you did it so that we can execute query from WASM without direct wsv access... But it still doesn't make sense to me from logic side of view